### PR TITLE
add launch_ps module so that we can launch a parameter server trainin…

### PR DESCRIPTION
**Pull request by guru4elephant**
…g job

should be merged after https://github.com/PaddlePaddle/Paddle/pull/20860 is merged.

1) a user can specify worker_num and server_num
2) parameter server can be killed after all workers exit
3) unit test is added
test=develop

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/fleet_ps_training.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+optimizer = fleet.distributed_optimizer(optimizer)
+optimizer.minimize(cost)
+
+if fleet.is_server():
+    fleet.init_server()
+    fleet.run_server()
+elif fleet.is_worker():
+    place = fluid.CPUPlace()
+    exe = fluid.Executor(place)
+    exe.run(fluid.default_startup_program())
```
Comment (seiriosPlus) [inline python/paddle/fluid/tests/unittests/fleet_ps_training.py]: fleet.statup_program

File: python/paddle/distributed/launch_ps.py [hunk 2]
```diff
@@ -142,16 +138,20 @@ def start_procs(args):
...
 
-    for i in range(0, len(procs)):
-        proc = procs[i]
-
-        proc.wait()
+    # only wait worker to finish here
+    for i, proc in enumerate(procs):
+        if i < server_num:
+            continue
+        procs[i].wait()
```
Comment (seiriosPlus) [inline python/paddle/distributed/launch_ps.py:145]: 是否需要加 trainer防 hang 住检查？

Comment (guru4elephant) [reply python/paddle/distributed/launch_ps.py:145]: @seiriosPlus wait超时即杀掉？

**Review by seiriosPlus — APPROVED** [review]
LGTM

**Pull request merged**